CONTRIBUTING.md

```diff
@@ -1,60 +1,357 @@
-# How to contribute
+#  Contributing to horizon
+So, you want to hack on horizon? Yay!
 
-## Welcome
+First of all, thank you for considering contributing to our project! We appreciate your time and effort, and we value any contribution, whether it's reporting a bug, suggesting a new feature, or submitting a pull request.
 
-Welcome to HorizonCD, and thanks for your interest in contributing to us!
+This document provides guidelines and best practices to help you contribute effectively.
 
-Below is a set of things you should know before you open a PR.
+## Requirements
 
-## Issues
+Please meet the minimum version of the Go language published in [go.mod](./go.mod). If you want to manage the Go language version, we provide tools to install [gvm](https://github.com/moovweb/gvm) in our [Makefile](./Makefile)
 
-Opening an issue is a great way to contribute to this project.
+You'd better use Linux as the development environment, Linux with [Makefile](./Makefile) can help you quickly build and test horizon project.
 
-Before opening an issue, you should always check if there's one containing your good thought exists.
+If you are familiar with [Makefile](./Makefile) , you can easily see the clever design of the horizon Makefile. Storing the necessary tools such as golangci in the `/tools` directory can avoid some tool version issues.
 
-If there's feedback\comments or anything you are not sure about, feel free to open an issue, whenever day and night.
+The [Makefile](./Makefile) is for every developer, even if you don't know how to use the Makefile tool, don't worry, we provide two great commands to get you up to speed with the Makefile architecture, `make help` and `make help-all`, it can reduce problems of the developing environment.
 
-To make the issue details as standard as possible, we setup an [ISSUE TEMPLATE](https://github.com/horizoncd/horizon/tree/main/.github/ISSUE_TEMPLATE) for issue reporters. You can find two kinds of issue templates there: bug report and feature request. Please BE SURE to follow the instructions to fill fields in template.
-An issue about bugs should tell us:
 
-There are a lot of cases when you could open an issue:
+## Code of Conduct
+
+We divide the problem into security and general problems:
+
+#### Reporting security issues
+
+Security issues are always treated seriously. As our usual principle, we discourage anyone to spread security issues. If you find a security issue of horizon, please do not discuss it in public and even do not open a public issue.
+
+You can also message us on [slack]((https://join.slack.com/t/horizoncd/shared_invite/zt-1pqpobiwn-1mgV60SIa1oi4mL1WXf7uA)) or chat with us on wechat.
```

Comment:
this link expired, replace with https://join.slack.com/t/horizoncd/shared_invite/zt-1sehbmzcx-dgIwaExNR4fZKXppj5kmgQ